spring-cloud-azure-autoconfigure/src/main/java/com/microsoft/azure/spring/cloud/autoconfigure/telemetry/TelemetrySender.java

```diff
@@ -5,43 +5,56 @@
  */
 package com.microsoft.azure.spring.cloud.autoconfigure.telemetry;
 
-import com.microsoft.applicationinsights.TelemetryClient;
+import com.fasterxml.jackson.databind.ObjectMapper;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
+import org.springframework.http.*;
 import org.springframework.lang.NonNull;
 import org.springframework.util.StringUtils;
+import org.springframework.web.client.RestTemplate;
 
 import java.util.concurrent.Executors;
 import java.util.concurrent.ScheduledExecutorService;
 import java.util.concurrent.TimeUnit;
 
+import static org.springframework.util.MimeTypeUtils.APPLICATION_JSON;
+
 public class TelemetrySender {
     private static final Logger log = LoggerFactory.getLogger(TelemetrySender.class);
 
     private static final int INSTRUMENTATION_KEY_LENGTH = 36;
 
-    private final TelemetryClient client;
+    private static final String TELEMETRY_TARGET_URL = "https://dc.services.visualstudio.com/v2/track";
```

Comment:
Could you move all rest client logic into a class such as TelemetryRestClient, then just replace original client into this one? In the future, we just need to replace this client.

Reply:
I use the same name with TelemetrySender for that purpose in other repo, but you have this name already. Will try new name for this.